soccer/gameplay/evaluation/passing_positioning.py

```diff
@@ -0,0 +1,175 @@
+
+            # Force algorithm not to check within a min distance
+            if ((candidate - pos).mag() > min_dist):
+                outlist.extend([candidate])
+            currentx += step
+
+        currenty += step
+        currentx = rect.min_x()
+
+    return outlist
+
+
+## Evaluates a single point and returns the overall coefficient for the area
+#
+# @param kick_point: Point where we are kicking from
+# @param receive_point: Point to which are kicking to
+# @param ignore_robots: Robots to ignore
+# @param field_weights: A tuple of the 3 difference weights to apply to field position 
+#               (Centerness, Distance to their goal, Angle off their goal)
+# @param weights: A tuple of the 4 different weights to apply to the evaulations overall
+#               (space, field_position, shot_chance, kick_proximty)
+# @return Returns a score between 0 and 1 on how good of pass would be
+def eval_singl_point(kick_point, receive_point, ignore_robots, field_weights,
+                     weights):
+
+    if kick_point is None:
+        if main.ball().valid:
+            kick_point = main.ball().pos
+        else:
+            return None
+
+    shotChance = 0
+
+    # Dissallow shooting over midfield
+    if (kick_point.y > constants.Field.Length / 2):
+        # TODO: Replace with KickEval
+        shotChance = evaluation.shooting.eval_shot(receive_point)
+
+    # TODO: Replace with KickEval
+    passChance = evaluation.passing.eval_pass(kick_point, receive_point)
+
+    space = evaluation.field.space_coeff_at_pos(receive_point, ignore_robots)
+    fieldPos = evaluation.field.field_pos_coeff_at_pos(
+        receive_point, field_weights[0], field_weights[1], field_weights[2])
+    distance = math.exp(-1 * (kick_point - receive_point).mag())
+
+    # All of the other scores are based on whether the pass will actually make it to it
+    # Not worth returning a great position if we cant even get a pass there
+    totalChance = passChance * ( weights[0] * (1 - space) + \
+                                 weights[1] * fieldPos + \
+                                 weights[2] * shotChance + \
+                                 weights[3] * (1 - distance) )
+
+    return totalChance / math.fsum(weights)
+
+
+## Finds the best position to pass to
+#
+# @param kick_point: Point that we are passing from
+# @param evaluation_zone: Area to evaluate passes to
+# @param ignore_robots: Robots to ignore when calculating scores
+# @param field_weights: A tuple of the 3 difference weights to apply to field position 
+#               (Centerness, Distance to their goal, Angle off their goal)
+# @param weights: A tuple of the 4 different weights to apply to the evaulations overall (Weights are normalized)
+#               (space, field_position, shot_chance, kick_proximity)
+# @param debug: Displays pass lines and scores onto the field when true
+# @return bestPoint and bestScore in that order
+def eval_best_receive_point(kick_point,
```

Comment:
Once this is merged in, we should probably replace the touchpass receive point in favor of this one so we don't have duplicates. We'll think about that later though.